File: RootTools2/src/main/java/com/rarnu/tools/neo/base/BaseDialog.java
package com.rarnu.tools.neo.base;

import android.app.Activity;
import android.app.Fragment;
import android.os.Bundle;

public abstract class BaseDialog extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        if (customTheme() != 0) {
            setTheme(customTheme());
        }
        super.onCreate(savedInstanceState);
        if (getCondition()) {
            finish();
            return;
        }
        replace();
    }

    public void replace() {
        Fragment bf = replaceFragment();
        getFragmentManager().beginTransaction().replace(android.R.id.content, bf).commit();
    }

    public abstract boolean getCondition();

    public abstract Fragment replaceFragment();

    public abstract int customTheme();
}

File: RootTools2/src/main/java/com/rarnu/tools/neo/base/BaseDialog.kt
package com.rarnu.tools.neo.base

import android.app.Activity
import android.app.Fragment
import android.os.Bundle

/**
 * Created by rarnu on 3/23/16.
 */
abstract class BaseDialog: Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        if (customTheme() != 0) {
            setTheme(customTheme())
        }
        super.onCreate(savedInstanceState)
        if (getCloseCondition()) {
            finish()
            return
        }
        replace()
    }

    open fun replace() {
        val bf = replaceFragment()
        fragmentManager.beginTransaction().replace(android.R.id.content, bf).commit()
    }

    abstract fun getCloseCondition(): Boolean
    abstract fun replaceFragment(): Fragment
    abstract fun customTheme(): Int
}